addons/website/static/src/components/configurator/configurator.js

```diff
@@ -639,11 +640,46 @@ async function applyConfigurator(self, themeName) {
             website_type: WEBSITE_TYPES[self.state.selectedType].name,
             logo_attachment_id: self.state.logoAttachmentId,
         };
-        const resp = await self.rpc({
+        let resp = await self.rpc({
             model: 'website',
             method: 'configurator_apply',
             kwargs: {...data},
         });
+        // If response asks for image processing: generate images and call again.
+        if (resp.imageProcessings) {
+            const doneProcessings = {};
+            for (const imgId in resp.imageProcessings) {
+                const imgAttrs = JSON.parse(resp.imageProcessings[imgId].replaceAll("'", '"'));
+                // Apply missing image processing.
+                const imgEl = document.createElement('img');
+                for (const attr in imgAttrs) {
+                    imgEl.setAttribute(attr, imgAttrs[attr]);
+                }
+                imgEl.src = imgEl.dataset.originalSrc;
+                delete imgEl.dataset.originalSrc;
+                await loadImageInfo(imgEl, self.rpc.bind(self));
+                const imgDataURL = await applyModifications(imgEl, {});
+                const newAttachmentSrc = await self.rpc({
+                    route: `/web_editor/modify_image/${imgEl.dataset.originalId}`,
+                    params: {
+                        data: imgDataURL.split(',')[1],
+                        mimetype: imgEl.dataset.mimetype,
+                        name: (imgEl.dataset.fileName ? imgEl.dataset.fileName : null),
+                    },
+                });
+                imgEl.src = newAttachmentSrc;
+                const updatedAttrs = imgEl.getAttributeNames().reduce((acc, name) => {
+                    return {...acc, [name]: imgEl.getAttribute(name)};
+                }, {});
+                doneProcessings[imgId] = updatedAttrs;
+            }
+            data['image_processings'] = doneProcessings;
```

Comment:
Maybe it could be processed in parallel as well ? With something like
```js
const keyValues = Object.entries(resp.imageProcessings);
const newImages = await Promise.all(keyValues.map(([imgId, rawData]) => generateImage(self, imgId, rawData)));
data['image_processings'] = Object.fromEntries(newImages);
```
With 
```js
async function generateImage(self, imgId, rawData) {
    const imgAttrs = JSON.parse(rawData.replaceAll("'", '"'));
    ...
    return [imgId, updatedAttrs];
}
```

Reply:
I also tried doing
```js
const newImages = await Promise.all(keyValues.map(async ([imgId, rawData]) => [imgId, await generateImage(self, rawData)]));
```
But I prefer your suggestion without this additional layer of async/await, so I kept `imgId` in the parameters.